pkg/quota/image/registry.go

```diff
@@ -12,11 +12,24 @@ import (
 // streams.
 func NewImageRegistry(osClient osclient.Interface) quota.Registry {
 	imageStream := NewImageStreamEvaluator(osClient)
+	return &generic.GenericRegistry{
+		InternalEvaluators: map[unversioned.GroupKind]quota.Evaluator{
+			imageStream.GroupKind(): imageStream,
+		},
+	}
+}
+
+// NewImageRegistry returns a registry for quota evaluation of OpenShift resources related to images and image
+// streams.
+func NewImageRegistryForAdmission(osClient osclient.Interface) quota.Registry {
```

Comment:
So the best that I can gather is that this is because of the fan-out problem where we are trying to quota the number of references to a non-namespaced resource.  This is something we never hit upstream because the closest parallel would be quota storage associated with a PV, but a PVC has a 1:1 relationship with a PV, whereas we have a 1:N relationship here...

I still think it would really help to have comments all over the place explaining this split.